DigitalLearningSolutions.Data/Services/UserService.cs

```diff
@@ -0,0 +1,70 @@
+﻿namespace DigitalLearningSolutions.Data.Services
+{
+    using System.Data;
+    using System.Linq;
+    using Dapper;
+    using DigitalLearningSolutions.Data.Models.User;
+
+    public interface IUserService
+    {
+        public User? GetUserByEmailAddress(string emailAddress);
+        public AdminUser? GetAdminUserByEmailAddress(string emailAddress);
+        public DelegateUser? GetDelegateUserByEmailAddress(string emailAddress);
+
+    }
+
+    public class UserService : IUserService
+    {
+        private readonly IDbConnection connection;
+
+        public UserService(IDbConnection connection)
+        {
+            this.connection = connection;
+        }
+
+        public User? GetUserByEmailAddress(string emailAddress)
+        {
+            // Assume user is a delegate (in Candidates Table)
+            User? user = GetDelegateUserByEmailAddress(emailAddress);
+
+            // If no Candidates entry found, look to AdminUsers
+            user ??= GetAdminUserByEmailAddress(emailAddress);
+
+            return user;
+        }
+
+        public AdminUser? GetAdminUserByEmailAddress(string emailAddress)
+        {
+            AdminUser? user = connection.Query<AdminUser>(
+                @"SELECT TOP (1)
+                        AdminID,
+                        Forename,
+                        Surname,
+                        Email,
+                        ResetPasswordID
+                    FROM AdminUsers
+                    WHERE (Email = @emailAddress)",
+                new { emailAddress }
+            ).FirstOrDefault();
```

Comment:
Do we need both FirstOrDefault and SELECT TOP (1)? Also, just heads up for next phases, but we'll need to get all delegate records in the future :P 

Another question I had was about getting admin user and getting delegate user being public methods - are these used anywhere else?